app/src/main/java/org/sopt/sample/SignUp/view/SignUpActivity.kt

```diff
@@ -0,0 +1,106 @@
+package org.sopt.sample.SignUp.view
+
+import android.content.Intent
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import org.sopt.sample.SignIn.view.SignInActivity
+import org.sopt.sample.SignUp.viewmodel.SignUpViewModel
+import org.sopt.sample.databinding.ActivitySignUpBinding
+
+class SignUpActivity : AppCompatActivity() {
+    private lateinit var binding: ActivitySignUpBinding
+    private val viewModel by viewModels<SignUpViewModel>()
+    private var idFlag = false
+    private var pwFlag = false
+    private var nameFlag = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignUpBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.btnSign.isEnabled = false
+        textWatcher()
+        clickEvent()
+    }
+
+    private fun idRegex(id : String) : Boolean{
+        return id.matches("^(?=.*[A-Za-z])(?=.*[0-9])[A-Za-z[0-9]]{6,10}$".toRegex())
+    }
+
+    private fun pwRegex(pw : String) : Boolean{
+        return pw.matches("^(?=.*[A-Za-z])(?=.*[0-9])(?=.*[!@#$%^&*?])[A-Za-z[0-9]!@#\$%^&*?]{6,12}$".toRegex())
+    }
+
+    private fun flagCheck(){
+        binding.btnSign.isEnabled = idFlag && pwFlag && nameFlag
+    }
+
+    private fun textWatcher(){
+        with(binding){
+            edtSignupId.addTextChangedListener(object : TextWatcher{
+                override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+                override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {}
+                override fun afterTextChanged(s: Editable?) {
+                    if(idRegex(edtSignupId.text.toString())){
+                        textinputlayoutId.error = null
+                        idFlag = true
+                    }
+                    else{
+                        textinputlayoutId.error = "아이디 형식이 올바르지 않습니다."
+                    }
+                    flagCheck()
+                }
+            })
```

Comment:
```suggestion
            edtSignupId.doAfterTextChanged {
                if (idRegex(edtSignupId.text.toString())){
                    textinputlayoutId.error = null
                    idFlag = true
                }
                else {
                    textinputlayoutId.error = "아이디 형식이 올바르지 않습니다."
                }
                flagCheck()
            }
        }
```

doAfterTextChanged 사용하면 좀 더 깔끔할 듯?